sdk/core/core-rest-pipeline/src/policies/formDataPolicy.ts

```diff
@@ -93,7 +93,6 @@ async function prepareFormData(formData: FormDataMap, request: PipelineRequest):
         });
       }
     }
-
-    request.multipartBody = { parts };
   }
+  request.multipartBody = { parts };
```

Comment:
How come this change was needed? Just want to make sure it's expected

Reply:
I think the idea here was to resolve a test issue where we were expecting this policy to set the body even when we passed in an empty `FormDataMap` -- maybe that doesn't make sense though? /cc @timovv to decide if we shuold only be setting this property if `formData` isn't empty

Reply:
I think the new behavior is more correct, good catch. My guess is I made a typo here and mixed up closing braces, and never noticed since everything still worked. 

I think setting the property is the correct behavior if formData is empty. The multipart layer should handle it just fine -- it will just produce a multipart body with no parts. We have a test case for multipartPolicy to make sure it handles that but maybe we should add one for formDataPolicy as well.

Reply:
@maorleger I'd be happy to approve a targeted PR that fixes form-data behavior if you want to toss it up. No worries about merge conflicts, I can merge it in after it hits main.

Reply:
Created https://github.com/Azure/azure-sdk-for-js/pull/28224 for this 

When reading formDataPolicy I also realized since we return early if we have no formData in the request this will result in no behavior changes, just avoids setting that field on every loop invocation. So hopefully a no-op pure refactor but please keep me honest